Dockerfile

```diff
@@ -1,4 +1,12 @@
-FROM python:3.10.7-alpine3.16 AS compile-stage
+# We use an Alpine base image in the compile-stage because of the build
+# requirements for some of the Python requirements. When the python3-dev
+# package is installed it will also install the python3 package which leaves us
+# with two Python installations if we use a Python Docker image. Instead we use
+# Alpine's python3 package here to create the virtual environment we will use
+# in the Python Docker image we use for the build-stage. The tag of the Python
+# Docker image matches the version of the python3 package available on Alpine
+# for consistency.
```

Comment:
Can the `3.17` be put in a variable, to DRY this out a bit?

Reply:
I'm sorry but what `3.17` are you referencing? If you're referencing `alpine:3.17` yes we could use a variable to reference the Alpine version being used, but then dependabot will be unable to do any dependency management for Docker images. It is also a door that I would like to close with regards to our Docker builds. I do not want to encourage changing Alpine version with build-time arguments. I would prefer that everything is as static as possible with build arguments only existing for things that explicitly should be configurable (like unprivileged user values).

Reply:
I understand but that will:
1. Break dependabot analysis
2. Encourage the idea that it's a configurable item which it is not intended to be due to both build reproducibility and strict version pinning of `apk` packages

Neither of which I am keen on for Docker configurations.

Reply:
> 2\. Encourage the idea that it's a configurable item which it is not intended to be due to both build reproducibility and strict version pinning of `apk` packages

This was not my intent, and a simple comment on the variable to dissuade folks from attempting to use it as a build variable would seem to suffice.  I just thought it would be helpful to DRY out, since I have been bitten a few times by only updating base image versions in one place with these multi-stage builds.